code/modules/awaymissions/redgate.dm

```diff
@@ -12,7 +12,8 @@
 	var/secret = FALSE	//If either end of the redgate has this enabled, ghosts will not be able to click to teleport
 	var/list/exceptions = list(
 		/obj/structure/ore_box,
-		/obj/structure/bed/roller
+		/obj/structure/bed/roller,
+		/obj/structure/bed/roller/adv
```

Comment:
I suggest removign roller/adv from the list and using is_type_in_list (or however proc name correctly goes) in the check for type.